realm/realm-library/src/androidTest/kotlin/io/realm/entities/DictionaryClass.kt

```diff
@@ -17,26 +17,50 @@
 package io.realm.entities
 
 import io.realm.Mixed
-import io.realm.MyDog
 import io.realm.RealmDictionary
 import io.realm.RealmObject
+import org.bson.types.Decimal128
+import org.bson.types.ObjectId
 import java.util.*
 
 open class DictionaryClass : RealmObject() {
 
-//    var myEmbeddedObjectDictionary: RealmDictionary<EmbeddedSimpleChild>? = null
-
-    // TODO: ObjectId and other Java primitives
-    var myDogDictionary: RealmDictionary<MyDog>? = null
-    var myUUIDDictionary: RealmDictionary<UUID>? = null
+    var myRealmModelDictionary: RealmDictionary<MyRealmModel>? = null
```

Comment:
Do we allow this to be nullable? I thought we checked for this in the annotation processor. But that shouldn't be the case. Just like lists are not allowed to be nu null this way.

Reply:
Created https://github.com/realm/realm-java/issues/7311 to handle it.